kotlinx-coroutines-core/common/src/CloseableCoroutineDispatcher.kt

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2016-2021 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.coroutines
+
+/**
+ * [CoroutineDispatcher] that provides method to close it,
+ * causing the rejection of any new tasks and cleanup of all underlying resources
+ * associated with the current dispatcher.
+ * Examples of closeable dispatchers are dispatchers backed by `java.lang.Executor` and
+ * by `kotlin.native.Worker`.
+ *
+ * **The `CloseableCoroutineDispatcher` class is not stable for inheritance in 3rd party libraries**, as new methods
+ * might be added to this interface in the future, but is stable for use.
+ */
+public expect abstract class CloseableCoroutineDispatcher() : CoroutineDispatcher {
+
+    /**
+     * Initiate the closing sequence of the coroutine dispatcher.
+     * After the successful call to [close], no new tasks will
+     * be accepted to be [dispatched][dispatch], but previously dispatched tasks will be run.
+     *
+     * Invocations of `close` on already closed dispatcher have no effect.
```

Comment:
```suggestion
     * Invocations of `close` on an already closed dispatcher have no effect.
```

Reply:
There's ambiguity: it is stated above that `close` *initiates* the closing sequence, but doesn't necessarily *close* the dispatcher. I assume the two are considered equivalent here? If so, two ways to rephrase this that I see:
* Talk here not about closed dispatchers, but only about those "that already entered the closing sequence". Cumbersome, in my opinion.
* Clarify via a more formal additional sentence that calls to `close` are *idempotent*.